datamodels/core/src/main/java/net/fhirfactory/pegacorn/core/file/common/DateUtils.java

```diff
@@ -0,0 +1,184 @@
+    public static final String YYYY_MM_DD_T_HH_MM_SS_INPUT = "yyyy-MM-dd'T'HH:mm:ss";
+    public static final String YYYY_MM_DD_T_HH_MM_SS_Z_OUTPUT = "yyyy-MM-dd'T'HH:mm:ssZ";
+    public static final String YYYY_MM_DD_T_HH_MM_SS_ZOFFSET = "yyyy-MM-dd'T'HH:mm:ss[+-]hh:mm";    
+    public static final String YYYY_MM_DD = "yyyy-MM-dd";
+    public static final String DD_MM_YYYY = "dd/MM/yyyy";
+    public static final String DD_MM_YYYY_HYPHENS = "dd-MM-yyyy";
+    public static final String DD_MM_YYYY_HH_MM = "dd/MM/yyyy HH/mm";
+    public static final String YYYYMMDDHHMMSS = "yyyyMMddHHmmss";
+    public static final String YYYYMMDD_HHMMSS = "yyyyMMdd_HHmmss";
+       
+    public static final String DEFAULT_TIME_ZONE = "Australia/Sydney";
+
+    /**
+     * Converts a {@link String} representation of a timestamp to a {@link Date}
+     * using the supplied input format.
+     * 
+     * @param timestamp
+     * @param timestampFormatter
+     * @return
+     */
+    public static Date getDate(String timestamp, DateTimeFormatter timestampFormatter) {
+        Date date = null;
+        if (StringUtils.isNotBlank(timestamp)) {
+            LocalDateTime localDateTime = LocalDateTime.parse(timestamp, timestampFormatter);
+            date = getDateFromLocalDateTime(localDateTime);
+        }
+        logger.debug("getDate({}, {})={}", timestamp, timestampFormatter, date);
+        return date;
+    }
+
+    public static String getFormattedInstantWithTimeZone(Instant instant, DateTimeFormatter timestampFormatter) {
+        String dataUntilTimestampStr = convertInstantToDefaultTimeZone(instant).format(timestampFormatter);
+        int indexThirdLastCharacter = dataUntilTimestampStr.length() - 3;
+        if (dataUntilTimestampStr.charAt(indexThirdLastCharacter) != ':') {
+            dataUntilTimestampStr = dataUntilTimestampStr.substring(0, indexThirdLastCharacter + 1) + ":" + 
+                    dataUntilTimestampStr.substring(indexThirdLastCharacter + 1);
+        }
+        return dataUntilTimestampStr;
+    }
+        
+    public static Date getDateFrom(String timestamp, DateTimeFormatter timestampFormatter) {
+        if (StringUtils.isBlank(timestamp)) {
+            return new Date(2999 - 1900, 12 - 1, 31, 23, 59, 59);
+        }
+
+        Date date = getDateFromZonedDateTime(ZonedDateTime.parse(timestamp, timestampFormatter));
+        logger.debug("getDateFrom({})={}", timestamp, date);
+        return date;
+    }
+
+    public static Date getDateFromZonedDateTime(ZonedDateTime zonedDateTime) {
+        return Date.from(zonedDateTime.toInstant());
+    }
+
+    public static Date getDateFromLocalDateTime(LocalDateTime localDateTime) {
+        return getDateFromZonedDateTime(localDateTime.atZone(getZoneId()));
+    }
+
+    private static ZoneId getZoneId() {
+//      return ZoneId.systemDefault(); // Returned GMT
+      return ZoneId.of(DEFAULT_TIME_ZONE);
```

Comment:
Good for us but probably needs to be configurable if we want user outside of our time zone to use it